File: src/main/java/net/averkhoglyad/chess/manager/core/service/ProfileServiceImpl.java
package net.averkhoglyad.chess.manager.core.service;

import net.averkhoglyad.chess.manager.core.sdk.data.User;

import java.nio.file.Files;
import java.nio.file.Path;
import java.util.ArrayList;
import java.util.List;
import java.util.stream.Collectors;
import java.util.stream.Stream;

import static net.averkhoglyad.chess.manager.core.helper.ExceptionHelper.doStrict;

public class ProfileServiceImpl implements ProfileService {

    private final Path profilesFile;

    public ProfileServiceImpl(Path profilesFile) {
        this.profilesFile = profilesFile;
    }

    @Override
    public List<User> load() {
        List<User> users = new ArrayList<>();
        if (Files.exists(profilesFile)) {
            try (Stream<String> lines = doStrict(() -> Files.lines(profilesFile))) {
                lines.map(User::new).forEachOrdered(users::add);
            }
        }
        return users;
    }

    @Override
    public void save(List<User> profiles) {
        List<String> usernames = profiles.stream()
            .map(User::getUsername)
            .collect(Collectors.toList());
        doStrict(() -> Files.write(profilesFile, usernames));
    }

}

File: src/main/kotlin/net/averkhoglyad/chess/manager/core/service/ProfileServiceImpl.kt
//package net.averkhoglyad.chess.manager.core.service
//
//import net.averkhoglyad.chess.manager.core.sdk.data.User
//
//import java.nio.file.Files
//import java.nio.file.Path
//import java.util.ArrayList
//import java.util.stream.Collectors
//import java.util.stream.Stream
//
//import net.averkhoglyad.chess.manager.core.helper.ExceptionHelper.doStrict
//
//class ProfileServiceImpl(private val profilesFile: Path) : ProfileService {
//
//    override fun load(): List<User> {
//        val users = ArrayList<E>()
//        if (Files.exists(profilesFile)) {
//            doStrict { Files.lines(profilesFile) }.use({ lines -> lines.map(Function<String, R> { User() }).forEachOrdered(Consumer<R> { users.add() }) })
//        }
//        return users
//    }
//
//    override fun save(profiles: List<User>) {
//        val usernames = profiles.stream()
//                .map(Function<User, R> { User.getUsername() })
//                .collect(Collectors.toList<Any>())
//        doStrict { Files.write(profilesFile, usernames) }
//    }
//
//}
